pkg/helpers/helpers.go

```diff
@@ -0,0 +1,147 @@
+package helpers
+
+import (
+	"fmt"
+
+	mcfgv1 "github.com/openshift/api/machineconfiguration/v1"
+	v1 "github.com/openshift/client-go/machineconfiguration/listers/machineconfiguration/v1"
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// osLabel is used to identify which type of OS the node has
+	OSLabel = "kubernetes.io/os"
+)
+
+func GetNodesForPool(mcpLister v1.MachineConfigPoolLister, nodeLister corev1listers.NodeLister, pool *mcfgv1.MachineConfigPool) ([]*corev1.Node, error) {
+	selector, err := metav1.LabelSelectorAsSelector(pool.Spec.NodeSelector)
+	if err != nil {
+		return nil, fmt.Errorf("invalid label selector: %w", err)
+	}
+
+	initialNodes, err := nodeLister.List(selector)
+	if err != nil {
+		return nil, err
+	}
+
+	nodes := []*corev1.Node{}
+	for _, n := range initialNodes {
+		p, err := GetPrimaryPoolForNode(mcpLister, n)
+		if err != nil {
+			klog.Warningf("can't get pool for node %q: %v", n.Name, err)
+			continue
+		}
+		if p == nil {
+			continue
+		}
+		if p.Name != pool.Name {
+			continue
+		}
+		nodes = append(nodes, n)
+	}
+	return nodes, nil
+}
+
+func GetPrimaryPoolForNode(mcpLister v1.MachineConfigPoolLister, node *corev1.Node) (*mcfgv1.MachineConfigPool, error) {
+	pools, err := GetPoolsForNode(mcpLister, node)
+	if err != nil {
+		return nil, err
+	}
+	if pools == nil {
+		return nil, nil
+	}
+	return pools[0], nil
+}
+
+func GetPoolsForNode(mcpLister v1.MachineConfigPoolLister, node *corev1.Node) ([]*mcfgv1.MachineConfigPool, error) {
```

Comment:
Here we are duplicating the code of this function https://github.com/openshift/machine-config-operator/pull/4000/files#diff-92f10e93215b8da12723d913e0985568114aab4b8afb67541c2a3a9ec5cafba6L661

Is it intended to duplicate the code?